src/lib/storage/segment_accessor.hpp

```diff
@@ -41,7 +41,7 @@ std::unique_ptr<AbstractSegmentAccessor<T>> create_segment_accessor(const std::s
  * accessor each.
  */
 template <typename T, typename SegmentType>
-class SegmentAccessor : public AbstractSegmentAccessor<T> {
+class SegmentAccessor final : public AbstractSegmentAccessor<T> {
```

Comment:
Did this have any performance impact or did you just add it for clarity and guideline reasons?

Reply:
In this case, it didn't. But since `final` is always a good thing, I didn't remove it again.